index.js

```diff
@@ -55,13 +125,17 @@ const htmlTarget3 = document.getElementById('a-3')
 htmlTarget3.innerHTML = answer3
 
 
-
 //------------------------ Question 4 ---------------------------
 // In the function below we are giving you an array of objects, each one with the same properties. Ask to the user for 3 diferentes options to sorting the array from the highest to lowest. In the case of a string, the criteria to sort must be the length of the string. The first one is sorting the array of objects based on the title property.
 // The second one sorting the array of objects based on the author property, the third one based on the library property. finally, the return value has to be the string sorted of the property selected separeted with a semicolon. Remember you have to sort all of the array based on the selected property
 //example: if the user select sorting by title the return value must be: "Mockingjay: The Final Book of The Hunger Games; Walter Isaacson; The Road Ahead"
 
-sort = () => {
+/* Doesn't work
+
+let sortedLibrary = [];
+let searchBy = prompt("You can sort it by 'title', 'author' and 'libraryID'. How do you want to sort it?");
+
+sort = (searchBy) => {
```

Comment:
for this function, we are trying to sort the objects in the `library` array in ascending order based on the "search by" parameter provided by the user.

so, if the user enters `"title"` we should see this result:
```js
[
   {
       title: 'Mockingjay: The Final Book of The Hunger Games',
       author: 'Suzanne Collins',
       libraryID: 3245
   },
   {
       title:  'The Road Ahead',
       author: 'Bill Gates',
       libraryID: 1254
   },
   {
       title: 'Walter Isaacson',
       author: 'Steve Jobs',
       libraryID: 4264
   }
]
```

so your `sort` function should look something like this:

```js
const sort = (sortBy) => {
  // create a new variable called `sortedLibrary`
  let sortedLibrary;

  // set the value of `sortedLibrary` to a custom sorted version of `library` array
  sortedLibrary = library.sort((a, b) => {
    // create a variable called `sortA` which is equal to `a.sortBy`
    // create a variable called `sortB` which is equal to `b.sortBy`
    // return `sortB` minus `sortA`
  }

  // return `sortedLibrary` (use JSON.stringify() to convert the array to a string)
}
```

here's some more information about JavaScript array sorting and compare functions: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort